ada_2_fhir/zibs2017/payload/zib-LaboratoryTestResult-Specimen-2.1.xsl

```diff
@@ -136,9 +136,7 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
                     <xsl:when test="specimen_source | bron_monster">
                         <xsl:variable name="profileValue">http://nictiz.nl/fhir/StructureDefinition/zib-MedicalDeviceProduct</xsl:variable>
                         <Device>
-                            <xsl:if test="string-length($containedDeviceId) gt 0">
-                                <id value="{nf:make-fhir-logicalid(tokenize($profileValue, './')[last()], $containedDeviceId)}"/>
-                            </xsl:if>
+                            <id value="{$containedDeviceId}"/>
```

Comment:
Is the removed check on the length of the id not needed to ensure the `.id` is valid?

Reply:
`nf:make-fhir-logicalid(tokenize($profileValue, './')[last()], $containedDeviceId)` was not correct, so that has been changed to `$containedDeviceId`.

I removed the `if` check because the code only lands here when `test="specimen_source | bron_monster"`. If that is the case, `$containedDeviceId` will always be filled by definition. So imho the check doesn't do anything.